### PR TITLE
Update enum_state_machine for ghenghis Ruby 3 upgrade

### PR DESCRIPTION
**Pull request by GabrielaJustine16**
## Description

This PR is 5 of 7 gem changes for the Upgrade of Ruby happening in [Ghenghis](https://github.com/CityBaseInc/ghenghis). Ruby 2.7.6 is being upgraded to Ruby 3.0.5. 

**_Positional and keyword arguments will now be separated._**

[PLFM-109](https://citybase.atlassian.net/browse/PLFM-109)
## Changes

- [x]  Changes made to support the way keyword arguments are now handled

## Guidelines

[PR Author and Reviewer guidelines](https://citybase.atlassian.net/wiki/spaces/DEV/pages/549322760/Code+Review+Guidelines)

## Concerns

(optional)

## Justification for no test coverage

(optional)

## Dependencies (other repos or epic branch)

[enQuesta_api](https://github.com/CityBaseInc/enQuesta_api/pull/16)
[forte_api](https://github.com/CityBaseInc/forte_api/pull/8)
[indy_marion_county_treasurer_api](https://github.com/CityBaseInc/indy_marion_county_treasurer_api/pull/4)
[tyler_new_world_api](https://github.com/CityBaseInc/tyler_new_world_api/pull/14)
[enum_state_machine](https://github.com/HornsAndHooves/enum_state_machine/pull/10)
[coc_crs_api](https://github.com/CityBaseInc/coc_crs_api/pull/32)
[power_enum_2](https://github.com/albertosaurus/power_enum_2/pull/27)

## Screenshots

(optional)

### CODE REVIEW

File: lib/enum_state_machine/integrations/active_record.rb [hunk 1]
```diff
@@ -486,11 +486,11 @@ def initialize(*)
         def define_action_hook
           if action_hook == :save
             define_helper :instance, <<-end_eval, __FILE__, __LINE__ + 1
-              def save(*)
+              def save(**)
```
Comment (albertosaurus) [inline lib/enum_state_machine/integrations/active_record.rb:489]: @GabrielaJustine16 What's the minimum Ruby version that supports this syntax?

Comment (baburdick) [reply lib/enum_state_machine/integrations/active_record.rb:489]: Double-splat goes back to Ruby 2.0.

**Comment by baburdick** [comment]
Thanks @samgranieri. @albertosaurus is on offline this week. The work you're doing is of great interest to us, as we're also gearing up to move to Ruby 3+. I hope to look more thoroughly in the next couple nights. If your timeline is more urgent, you may have to reference your CityBaseInc fork until we all agree this is ready to release. Thanks again to you both for your extensive work on this.

**Comment by jwpincus** [comment]
No rush since we can operate off a branch, but just curious if you guys have a rough timeline for merging this in? @albertosaurus, @baburdick

**Comment by albertosaurus** [comment]
Hopefully this week. We're a bit swamped right now.

**Comment by samgranieri** [comment]
@albertosaurus have things cleared up a bit for you? We at CityBase are keen to contribute this back to open source on rubygems.org instead of relying on a fork

**Comment by maneyko** [comment]
@GabrielaJustine16 @samgranieri @jwpincus, please pull in the latest revision of `master`. Almost all of your proposed changes to `lib/` and `test/` were incorporated in #11 and are in [`v0.8.0`](https://github.com/HornsAndHooves/enum_state_machine/tree/v0.8.0).

**Comment by baburdick** [comment]
@albertosaurus, @maneyko: Is this now moot?

**Comment by maneyko** [comment]
Yes, this can be closed. The changes to `lib/` are identical and the only difference between the changes to `test/` is that this PR removes the `EvalHelpersSymbolTaintedMethodTest` class — which probably should have been removed in #11 since it now does nothing without `taint`.

**Pull request closed without merging**